pkg/operator/ceph/disruption/clusterdisruption/osd.go

```diff
@@ -168,109 +199,125 @@ func (r *ReconcileClusterDisruption) reconcilePDBsForOSDs(
 			logger.Debugf("Ceph %q cluster not ready, cannot check Ceph status yet.", request.Namespace)
 			return nil
 		}
-		return errors.Wrapf(err, "could not check cluster health")
+		return errors.Wrapf(err, "failed to check cluster health")
+	}
+
+	drainingFailureDomainList := getSortedOSDMapKeys(drainingFailureDomainsMap)
+	activeDrains := len(drainingFailureDomainList) > 0
+
+	// create blocking PDBs for all the OSDs that are not being drained right now
+	if activeDrains {
+		logger.Infof("pg health: %q. detected drains on %q: %v", pgHealthMsg, poolFailureDomain, drainingFailureDomainList)
+		for fd, osds := range allFailureDomainsMap {
+			for _, osd := range osds {
```

Comment:
Ah, here I think we just need to create the failure-domain PDBs instead of also iterating through each of the OSDs. Note that we may need to add labels to the OSD deployments with their failure domain info, not sure we have it today.

Reply:
we determine the failure domain after the pools are created. So they are not in the OSD pods.

Reply:
adding the crush location hierarchy as labels in the OSD deployment to create the PDB.  Example 
```
labels:
    app: rook-ceph-osd
    ceph-osd-id: "1"
    ceph.rook.io/DeviceSet: ocs-deviceset-gp2-0
    ceph.rook.io/pvc: ocs-deviceset-gp2-0-data-0-24zjz
    ceph_daemon_id: "1"
    ceph_daemon_type: osd
    **crush-location-host: ocs-deviceset-gp2-0-data-0-24zjz
    crush-location-region: us-east-1
    crush-location-root: default
    crush-location-zone: us-east-1b**
    failure-domain: ocs-deviceset-gp2-0-data-0-24zjz
    osd: "1"
    pod-template-hash: 854c695bb7
    portable: "true"
    rook_cluster: openshift-storage
```

Reply:
How about calling the labels `topology-location-host`, `topology-location-zone`, etc?

Reply:
updated labels to use `topolocy-location-<>` format.  Example: 
 labels:
    app: rook-ceph-osd
    ceph-osd-id: "0"
    ceph_daemon_id: "0"
    ceph_daemon_type: osd
    failure-domain: minikube
    osd: "0"
    pod-template-hash: 777f5b95c7
    portable: "false"
    rook_cluster: rook-ceph
    **topology-location-host: minikube
    topology-location-root: default**